app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/registration/SubscriptionRecoveryController.scala

```diff
@@ -76,6 +81,14 @@ class SubscriptionRecoveryController @Inject() (
         result.flatMap(identity)
     }
 
+  def eoriExist(service: Service, journey: Journey.Value): Action[AnyContent] =
+    authAction.ggAuthorisedUserWithServiceAction {
+      implicit request => _: LoggedInUserWithEnrolments =>
+        for {
+          eori <- sessionCache.eori
+        } yield Ok(alreadyHaveEori(eori))
```

Comment:
Can be one liner `sessionCache.eori.map(Ok(alreadyHaveEori(_)))` but the current implementation is good too.

Reply:
True - just copy/pasted from ROSM frontend.